### PR TITLE
Convert monster to latest widget

### PR DESCRIPTION
**Pull request by agubler**

### CODE REVIEW

File: monster-cards/src/app.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import createProjector, { Projector } from 'dojo-widgets/createProjector';
+
+import createNavbar from './widgets/navbar/createNavbar';
+import createHomePage from './widgets/home/createHomePage';
+import createCardDetailsPage from './widgets/card-details/createCardDetailsPage';
+import createCardsPage from './widgets/cards/createCardsPage';
+import createGameplayPage from './widgets/gameplay/createGameplayPage';
+import createAboutPage from './widgets/about/createAboutPage';
+
+function getPageRoute(instance: any) {
```
Comment (Tomdye) [inline monster-cards/src/app.ts]: Perhaps rename to `getPageFromRoute` and change the `let route` to `let page` as you're returning a page, not a route.

Comment (Tomdye) [inline monster-cards/src/app.ts]: can `instance` be typed? `Widget` ?

File: monster-cards/src/app.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	let route: any;
```
Comment (Tomdye) [inline monster-cards/src/app.ts]: again, this can be type I believe: `DNode`

File: monster-cards/src/index.html [hunk 3]
```diff
@@ -4,9 +4,6 @@
 		<meta charset="utf-8">
 		<title>dojo2 Monster Cards</title>
 	</head>
-	<body is="app-projector" data-css-transitions>
```
Comment (Tomdye) [inline monster-cards/src/index.html]: You need a document body.

File: monster-cards/src/main.ts [hunk 4]
```diff
@@ -1,58 +1,18 @@
...
-import createCardDetailsPage from './widgets/card-details/createCardDetailsPage';
-import defaultWidgetStore from './stores/widgetStore';
-
+import { bindActions as bindCardStoreActions } from './stores/cardStore';
+import widgetStore from './stores/widgetStore';
+import createApp from './app';
 import 'maquette/src/css-transitions';
 
-const app = createApp({ defaultWidgetStore });
+const root = document.getElementsByTagName('my-app')[0];
```
Comment (Tomdye) [inline monster-cards/src/main.ts]: why not just place at `document.body` in this case?

File: monster-cards/src/index.html [hunk 5]
```diff
@@ -4,9 +4,6 @@
...
-		<header is="app-widget" data-uid="navbar"></header>
-		<main is="app-widget" data-uid="container"></main>
-	</body>
+	<my-app></my-app>
```
Comment (Tomdye) [inline monster-cards/src/index.html]: if not just placing at `document.body` a better name would likely be useful, ie `<monster-app>`

File: monster-cards/src/widgets/card-details/createCardNavBar.ts [hunk 6]
```diff
@@ -15,17 +15,15 @@ const createCardNavBar = createWidgetBase
 	.mixin({
 		mixin: {
 			classes: [ 'animated', 'cardNavBar' ],
-			childNodeRenderers: [
-				function(this: CardNavBar): DNode[] {
-					const { cards } = this.state;
+			getChildrenNodes: function(this: CardNavBar): DNode[] {
+				const { cards = [] } = this.state;
```
Comment (Tomdye) [inline monster-cards/src/widgets/card-details/createCardNavBar.ts]: good catch 👍

File: monster-cards/src/widgets/cards/createCardsList.ts [hunk 7]
```diff
@@ -13,15 +13,16 @@ const createCardList = createWidgetBase
 	.mixin({
 		mixin: {
 			classes: [ 'cardList' ],
-			childNodeRenderers: [
-				function(this: CardList): DNode[] {
-					const { cards } = this.state;
+			getChildrenNodes: function(this: CardList): DNode[] {
+				const { cards } = this.state;
```
Comment (Tomdye) [inline monster-cards/src/widgets/cards/createCardsList.ts]: could default to `[]` as you have done elsewhere for consistency, and then the `if` would not be required.

File: monster-cards/src/widgets/navbar/createNavbar.ts [hunk 8]
```diff
@@ -14,33 +14,37 @@ export type NavBarLinkDefinition = {
...
+let counter = 0;
+
+function listItem(childNode: DNode, key?: any): DNode {
+	const options = key ? { key } : {};
+	return d('li', options, [ childNode ]);
 }
 
 function createNavBarLink({ text: innerHTML, href }: NavBarLinkDefinition): DNode {
-	return listItem(d('a', { href, innerHTML }));
+	return listItem(d('a', { key: counter++, href, innerHTML }), innerHTML);
```
Comment (Tomdye) [inline monster-cards/src/widgets/navbar/createNavbar.ts]: did this get changed to `key` now? last time I checked we were still using `id` here.

File: monster-cards/src/main.ts (hunk 4 above)
Comment (agubler) [inline monster-cards/src/main.ts]: showing an example of attaching not to the default node.

File: monster-cards/src/widgets/navbar/createNavbar.ts (hunk 8 above)
Comment (agubler) [inline monster-cards/src/widgets/navbar/createNavbar.ts]: `key` is Maquettes API - this is creating a `HNode` so is a `VNodeProperties` object.

File: monster-cards/src/index.html (hunk 3 above)
Comment (agubler) [inline monster-cards/src/index.html]: turns out you don't (as this works) but I didn't actually mean to remove the `body`. An oversight on culling the `index.html`

File: monster-cards/src/app.ts (hunk 1 above)
Comment (agubler) [inline monster-cards/src/app.ts]: Yup, it is actually `Projector`

File: monster-cards/src/widgets/cards/createCardsList.ts (hunk 7 above)
Comment (agubler) [inline monster-cards/src/widgets/cards/createCardsList.ts]: nice, yup.

File: monster-cards/src/main.ts (hunk 4 above)
Comment (agubler) [inline monster-cards/src/main.ts]: done already.

**Pull request merged**